### PR TITLE
[FLINK-31136][sql-client] Forbid sql client to read flink-conf.yaml i…

### PR DESCRIPTION
**Pull request by fsk119**
…n the gateway mode

## What is the purpose of the change

*In the client mode, sql client should not read the execution config, which may have different configuration with gateway configuration.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java [hunk 1]
```diff
@@ -142,11 +151,13 @@ public static DefaultContext load(
         FileSystem.initialize(
                 configuration, PluginUtils.createPluginManagerFromRootFolder(configuration));
 
-        if (discoverPythonDependency) {
-            dependencies = new ArrayList<>(dependencies);
-            dependencies.addAll(discoverPythonDependencies());
+        if (!discoverExecutionConfig) {
```
Comment (LadyForest) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java:154]: I felt that the naming `discoverExecutionConfig` is a little hard to understand, why this param can control the python dependency discovery.

**Review by LadyForest — COMMENTED** [review]
Thanks for the contribution. It would be better to add a test verifying the fix.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java (hunk 1 above)
Comment (fsk119) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java]: Ok. I add another parameter to manage the behavior separately.

**Review by LadyForest — APPROVED** [review]
Thanks for the update, LGTM

**Pull request merged**